crates/oxc_linter/src/rules/unicorn/prefer_math_min_max.rs

```diff
@@ -0,0 +1,257 @@
+    /// Examples of **correct** code for this rule:
+    /// ```javascript
+    /// Math.min(height, 50);
+    /// Math.max(height, 50);
+    /// ```
+    PreferMathMinMax,
+    pedantic,
+    fix
+);
+
+impl Rule for PreferMathMinMax {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::ConditionalExpression(conditional_expr) = node.kind() else {
+            return;
+        };
+
+        let Expression::BinaryExpression(test_expr) = &conditional_expr.test else {
+            return;
+        };
+
+        let condition_type =
+            is_min_max(&test_expr, &conditional_expr.consequent, &conditional_expr.alternate, &ctx);
+
+        if matches!(condition_type, TypeOptions::Unknown) {
+            return;
+        }
+
+        ctx.diagnostic_with_fix(prefer_math_min_max_diagnostic(conditional_expr.span), |fixer| {
+            let Some(consequent) = get_expr_value(&conditional_expr.consequent) else {
+                return fixer.noop();
+            };
+            let Some(alternate) = get_expr_value(&conditional_expr.alternate) else {
+                return fixer.noop();
+            };
+
+            return match condition_type {
+                TypeOptions::Max => fixer
+                    .replace(conditional_expr.span, format!("Math.max({consequent}, {alternate})")),
+                TypeOptions::Min => fixer
+                    .replace(conditional_expr.span, format!("Math.min({consequent}, {alternate})")),
+                _ => fixer.noop(),
+            };
+        });
+    }
+}
+
+fn get_expr_value(expr: &Expression) -> Option<String> {
+    return match expr {
+        Expression::NumericLiteral(lit) => Some(lit.to_string()),
+        Expression::UnaryExpression(lit) => {
+            let mut unary_str: String = String::from(lit.operator.as_str());
+            let Some(unary_lit) = get_expr_value(&lit.argument) else {
+                return Some(unary_str.to_string());
+            };
+
+            unary_str.push_str(unary_lit.as_str());
+
+            return Some(unary_str.to_string());
+        }
+        Expression::Identifier(identifier) => Some(identifier.name.to_string()),
+        _ => None,
+    };
+}
+
+fn is_same_expression(left: &Expression, right: &Expression, ctx: &LintContext) -> bool {
+    if is_same_reference(left, right, ctx) {
+        return true;
+    }
+
+    return match (left, right) {
+        (Expression::UnaryExpression(left_expr), Expression::UnaryExpression(right_expr)) => {
+            left_expr.operator == UnaryOperator::UnaryNegation
+                && right_expr.operator == UnaryOperator::UnaryNegation
+                && is_same_reference(&left_expr.argument, &right_expr.argument, ctx)
+        }
+        _ => false,
+    };
+}
+
+fn is_min_max(
+    condition: &BinaryExpression,
+    consequent: &Expression,
+    alternate: &Expression,
+    ctx: &LintContext,
+) -> TypeOptions {
+    let is_matched = match (&condition.left, &condition.right) {
```

Comment:
```suggestion
    let is_matched = match (condition.left.get_inner_expression(), condition.left.get_inner_expression()) {
```
this allows cases where the consequent/alternate are wrapped in parenthesised/non null/as expressiont so be reported

Reply:
Good call out 👍, tweaked!